From 2df57798c7472ee79a3acb5be21ff1146231a36c Mon Sep 17 00:00:00 2001
From: Ke Qinghua <qinghua.ke@freescale.com>
Date: Fri, 7 Mar 2014 11:46:00 +0800
Subject: [PATCH 291/304] ENGR00302221-2:CTS test faild for security setting
 and test Vroot

Add graphics_device dir/file access rules for sabreauto_6q

Signed-off-by: Ke Qinghua <qinghua.ke@freescale.com>
---
 sabreauto_6q/BoardConfig.mk  | 5 +++++
 sabreauto_6q/sepolicy/app.te | 4 ++++
 2 files changed, 9 insertions(+)
 create mode 100755 sabreauto_6q/sepolicy/app.te

diff --git a/sabreauto_6q/BoardConfig.mk b/sabreauto_6q/BoardConfig.mk
index e6bce36..47c6a6a 100644
--- a/sabreauto_6q/BoardConfig.mk
+++ b/sabreauto_6q/BoardConfig.mk
@@ -103,3 +103,8 @@ endif
 endif
 
 TARGET_BOOTLOADER_CONFIG := mx6q:mx6q_sabreauto_android_config mx6dl:mx6dl_sabreauto_android_config mx6solo:mx6solo_sabreauto_android_config mx6q-nand:mx6q_sabreauto_nand_android_config mx6dl-nand:mx6dl_sabreauto_nand_android_config mx6solo-nand:mx6solo_sabreauto_nand_android_config
+BOARD_SEPOLICY_DIRS := \
+       device/fsl/sabreauto_6dq/sepolicy
+
+BOARD_SEPOLICY_UNION := \
+       app.te
diff --git a/sabreauto_6q/sepolicy/app.te b/sabreauto_6q/sepolicy/app.te
new file mode 100755
index 0000000..92d8fa6
--- /dev/null
+++ b/sabreauto_6q/sepolicy/app.te
@@ -0,0 +1,4 @@
+# Grant GPU access to all processes started by Zygote.
+# They need that to render the standard UI.
+allow domain    graphics_device:chr_file rw_file_perms;
+allow domain    graphics_device:dir { write create add_name setattr mounton search };
-- 
1.8.0

